src/lsdb/catalog/dataset/healpix_dataset.py

```diff
@@ -1480,3 +1481,59 @@ def plot_points(
             fig=fig,
             **kwargs,
         )
+
+    def estimate_size(self):
+        """Estimate size of catalog.
+
+        The result includes an upper-bound on the number of rows, and
+        a range of expected sizes for the data in memory and on disk.
+
+        This method should be used to estimate the size of a single catalog
+        when using column selection and spatial filtering. If the catalog
+        is modified further (especially involving other catalogs, e.g. joining
+        or crossmatching), the results will be innacurate.
+
+        The results are for RAW data, and they do not represent a full size
+        footprint, as the estimates provided by the catalog's parquet metadata
+        only account for the size of parquet data pages.
+        """
+        from human_readable import file_size, int_comma
+
+        warnings.warn("The estimates provided are approximations.")
+
+        def get_row_count(stats):
+            return stats.groupby(level=0).first()["row_count"].sum()
+
+        def get_mem_size(stats):
+            return pd.to_numeric(stats["memory_bytes"], errors="coerce").sum()
+
+        def get_disk_size(stats):
+            return pd.to_numeric(stats["disk_bytes"], errors="coerce").sum()
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            pixel_stats = self.per_pixel_statistics(
+                multi_index=True,
+                use_default_columns=False,
+                exclude_hats_columns=False,
+                include_columns=self.columns,
+                include_stats=("row_count", "disk_bytes", "memory_bytes"),
+            )
+
+        # Row count estimate
+        orig_cat = hc.read_hats(self.hc_structure.catalog_path)
+        expected_cat_rows = int(get_row_count(pixel_stats))
+        row_ratio = expected_cat_rows / int(orig_cat.catalog_info.total_rows) * 100
```

Comment:
```suggestion
        row_pct = expected_cat_rows / int(orig_cat.catalog_info.total_rows) * 100
```
Or another suffix that indicates percentage rather than ratio.